### PR TITLE
test: enhance general coverage

### PR DESCRIPTION
**Pull request by fzipi**
In the spirit of getting better coverage, this PR migrates tests to testify.

- move tests to testify
- achieve coverage > 80%

Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

### CODE REVIEW

File: check/base_test.go [hunk 1]
```diff
@@ -46,49 +44,33 @@ func TestNewCheck(t *testing.T) {
...
-		t.Error("Problem setting expected output")
-	}
+	assert.True(t, c.expected.ExpectError, "Problem setting expected output")
 
 	c.SetNoLogContains("nologcontains")
 
-	if c.expected.NoLogContains != "nologcontains" {
-		t.Error("PRoblem setting nologcontains")
-	}
+	assert.Equal(t, c.expected.NoLogContains, "nologcontains", "PRoblem setting nologcontains")
```
Comment (theseion) [inline check/base_test.go:51]:
```suggestion
	assert.Equal(t, c.expected.NoLogContains, "nologcontains", "Problem setting nologcontains")
```

File: check/logs_test.go [hunk 2]
```diff
@@ -16,24 +18,22 @@ var logText = `[Tue Jan 05 02:21:09.637165 2021] [:error] [pid 76:tid 1396834345
...
+	assert.NoError(t, err)
+
 	logName, _ := utils.CreateTempFileWithContent(logText, "test-*.log")
 	defer os.Remove(logName)
 	config.FTWConfig.LogFile = logName
 
 	c := NewCheck(config.FTWConfig)
 
 	c.SetLogContains(`id "920300"`)
+	assert.True(t, c.AssertLogContains(), "did not found expected content")
```
Comment (theseion) [inline check/logs_test.go:30]:
```suggestion
	assert.True(t, c.AssertLogContains(), "did not find expected content 'id \"920300\"'")
```

File: check/logs_test.go [hunk 3]
```diff
@@ -16,24 +18,22 @@ var logText = `[Tue Jan 05 02:21:09.637165 2021] [:error] [pid 76:tid 1396834345
...
 
-	// c.SetNoLogContains(`Something that is not there`)
-
-	if !c.AssertLogContains() {
-		t.Errorf("Failed !")
-	}
+	c.SetNoLogContains("SOMETHING")
+	assert.True(t, c.AssertNoLogContains(), "found something that is not there")
+	c.SetNoLogContains(`id "920300"`)
+	assert.False(t, c.AssertNoLogContains(), "did not found expected content")
```
Comment (theseion) [inline check/logs_test.go:37]:
```suggestion
	assert.False(t, c.AssertNoLogContains(), "did not find expected content")
```

File: check/response_test.go [hunk 4]
```diff
@@ -22,29 +24,22 @@ var expectedResponseFailTests = []struct {
...
-	if err != nil {
-		t.Errorf("Failed!")
-	}
 	c := NewCheck(config.FTWConfig)
 	for _, e := range expectedResponseOKTests {
 		c.SetExpectResponse(e.expected)
-		if !c.AssertResponseContains(e.response) {
-			t.Errorf("Failed !")
-		}
+		assert.True(t, c.AssertResponseContains(e.response), "response not expected")
```
Comment (theseion) [inline check/response_test.go:32]:
```suggestion
		assert.Truef(t, c.AssertResponseContains(e.response), "unexpected response: %v", e.response)
```

File: check/response_test.go [hunk 5]
```diff
@@ -22,29 +24,22 @@ var expectedResponseFailTests = []struct {
...
-	}
+	assert.NoError(t, err)
+
 	c := NewCheck(config.FTWConfig)
 	for _, e := range expectedResponseFailTests {
 		c.SetExpectResponse(e.expected)
-		if c.AssertResponseContains(e.response) {
-			t.Errorf("Failed !")
-		}
+		assert.False(t, c.AssertResponseContains(e.response), "response shouldn't contain text")
```
Comment (theseion) [inline check/response_test.go:43]:
```suggestion
		assert.Falsef(t, c.AssertResponseContains(e.response), "response shouldn't contain text %v", e.response)
```

File: config/config_test.go [hunk 6]
```diff
@@ -37,50 +39,33 @@ func TestNewConfigBadFileConfig(t *testing.T) {
...
-	if len(FTWConfig.TestOverride.Ignore) == 0 {
-		t.Errorf("Failed! Len must be > 0")
-	}
-
-	if reflect.ValueOf(FTWConfig.TestOverride.Input).IsZero() {
-		t.Errorf("Failed! Input must not be empty")
-	}
+	assert.NoError(t, err)
+	assert.Greater(t, len(FTWConfig.TestOverride.Ignore), 0, "Failed! Len must be > 0")
+	assert.False(t, reflect.ValueOf(FTWConfig.TestOverride.Input).IsZero(), "Failed! Input must not be empty")
```
Comment (theseion) [inline config/config_test.go:51]:
```suggestion
	assert.NotEmpty(t, reflect.ValueOf(FTWConfig.TestOverride.Input), "Failed! Input must not be empty")
```

File: config/config_test.go [hunk 7]
```diff
@@ -37,50 +39,33 @@ func TestNewConfigBadFileConfig(t *testing.T) {
...
 
 	for id, text := range FTWConfig.TestOverride.Ignore {
-		if !strings.Contains(id, "920400-1") {
-			t.Errorf("Looks like we could not find item to ignore")
-		}
-		if text != "This test result must be ignored" {
-			t.Errorf("Text doesn't match")
-		}
+		assert.True(t, strings.Contains(id, "920400-1"), "Looks like we could not find item to ignore")
```
Comment (theseion) [inline config/config_test.go:54]:
```suggestion
		assert.Contains(t, id, "920400-1", "Looks like we could not find item to ignore")
```

File: ftwhttp/request_test.go [hunk 8]
```diff
@@ -174,72 +156,56 @@ func TestRequestHeadersSet(t *testing.T) {
...
 	req := generateBaseRequestForTesting()
 
 	err := req.SetRawData([]byte("This is the data now"))
 
-	if err != nil && strings.Contains(err.Error(), "data field is already present in this request") {
-		t.Logf("Success !")
-	} else {
-		t.Errorf("Failed %s !", err.Error())
+	expectedError := errors.New("ftw/http: data field is already present in this request")
+	if assert.Error(t, err) {
```
Comment (theseion) [inline ftwhttp/request_test.go:188]: Why the `if`? Just write
```go
assert.Error(t, err)
assert.Equal(t, expectedError, err)
```
Or am I missing something?

Same for the instances below.

File: waflog/read_test.go [hunk 9]
```diff
@@ -133,17 +127,17 @@ func TestReadGetMarkedLinesWithTrailingEmptyLines(t *testing.T) {
 		foundLines[i], foundLines[j] = foundLines[j], foundLines[i]
 	}
 
-	if len(foundLines) != 6 {
-		t.Fatal("found unexpected number of log lines")
-	}
+	assert.Equal(t, len(foundLines), 6, "found unexpected number of log lines")
```
Comment (theseion) [inline waflog/read_test.go:130]:
```suggestion
	assert.Len(t, foundLines, 6, "found unexpected number of log lines")
```

File: waflog/read_test.go [hunk 10]
```diff
@@ -173,12 +166,82 @@ func TestReadGetMarkedLinesWithPrecedingLines(t *testing.T) {
 		foundLines[i], foundLines[j] = foundLines[j], foundLines[i]
 	}
 
-	if len(foundLines) != 4 {
-		t.Fatal("found unexpected number of log lines")
-	}
+	assert.Equal(t, len(foundLines), 4, "found unexpected number of log lines")
```
Comment (theseion) [inline waflog/read_test.go:169]:
```suggestion
	assert.Len(t, foundLines, 4, "found unexpected number of log lines")
```

**Review by theseion — CHANGES_REQUESTED** [review]

File: check/base_test.go (hunk 1 above)
Comment (fzipi) [reply check/base_test.go:51]: Changed.

File: ftwhttp/request_test.go (hunk 8 above)
Comment (fzipi) [reply ftwhttp/request_test.go]: I think I copied that from the testify readme 🤔 

Anyway, works and looks good without the if. Removed.

**Review by jcchavezs — APPROVED** [review]

File: config/config_test.go [hunk 11]
```diff
@@ -37,50 +38,33 @@ func TestNewConfigBadFileConfig(t *testing.T) {
...
-	if len(FTWConfig.TestOverride.Ignore) == 0 {
-		t.Errorf("Failed! Len must be > 0")
-	}
-
-	if reflect.ValueOf(FTWConfig.TestOverride.Input).IsZero() {
-		t.Errorf("Failed! Input must not be empty")
-	}
+	assert.NoError(t, err)
+	assert.Greater(t, len(FTWConfig.TestOverride.Ignore), 0, "Failed! Len must be > 0")
+	assert.NotEmpty(t, reflect.ValueOf(FTWConfig.TestOverride.Input), "Failed! Input must not be empty")
```
Comment (jcchavezs) [inline config/config_test.go:50]: Is this needed?

File: config/config_test.go [hunk 12]
```diff
@@ -37,50 +38,33 @@ func TestNewConfigBadFileConfig(t *testing.T) {
...
-
-	if len(FTWConfig.TestOverride.Ignore) == 0 {
-		t.Errorf("Failed! Len must be > 0")
-	}
-
-	if reflect.ValueOf(FTWConfig.TestOverride.Input).IsZero() {
-		t.Errorf("Failed! Input must not be empty")
-	}
+	assert.NoError(t, err)
+	assert.Greater(t, len(FTWConfig.TestOverride.Ignore), 0, "Failed! Len must be > 0")
```
Comment (jcchavezs) [inline config/config_test.go:49]: You can use https://pkg.go.dev/github.com/stretchr/testify/assert#NotEmpty
```suggestion
	assert.NotEmpty(t, FTWConfig.TestOverride.Ignore, "Ignore list must not be empty")
```

File: config/config_test.go (hunk 11 above)
Comment (fzipi) [reply config/config_test.go:50]: Probably not. Let me see if things work without it.

**Review by theseion — APPROVED** [review]

**Pull request merged**